frontend/test/metabase/scenarios/collections/collections.cy.spec.js

```diff
@@ -388,18 +388,11 @@ describe("scenarios > collection_defaults", () => {
         "**New collection should immediately be open, showing nested children**",
       );
 
-      openDropdownFor(NEW_COLLECTION);
-      cy.findAllByText("First collection");
-      cy.findAllByText("Second collection");
-
-      // TODO: This was an original test that made sure the collection is indeed open immediately.
-      //       That part is going to be addressed in a separate issue.
-      // cy.findByText(NEW_COLLECTION)
-      //   .closest("a")
-      //   .within(() => {
-      //     cy.icon("chevrondown");
-      //     cy.findByText("First collection");
-      //   });
+      getSidebarCollectionChildren(cy.findByText(NEW_COLLECTION)).within(() => {
```

Comment:
Yeah, I agree; it doesn't actually look like a Cypress test. Still, I think the helper might be useful in other collection page tests (e.g., some `permissions` tests query sidebar items).

The raw selector is pretty hard to read, and it isn't clear what it actually does (getting closest anchor element, then its parent, then its siblings). If we want to remove the function, I think we should at least leave a comment describing what's actually going on. Or we can slightly update the syntax, so it matches the Cypress style:

```js
getSidebarCollectionChildrenFor(NEW_COLLECTION)

function getSidebarCollectionChildrenFor(item) {
    return cy.findByText(item)
               .closest("a")
               .parent()
               .siblings();
}
```

Reply:
Where do parent and siblings come from all of a sudden?

Reply:
@kulyk I like the proposal for the updated syntax. Wanted to suggest something like that myself, but there are still 2 things here that can be improved.

1. If the sidebar collection is already open, this function will fail because `cy.findByText(item)` will find 2 strings - one in the sidebar, and one in the title. We can use new testid to prevent this:
```javascript
function getSidebarCollectionChildren(item) {
  return cy
    .findByTestId("sidebar")
    .findByText(item)
    .closest("a")
    .parent()
    .siblings();
}
```

2. Using `.siblings()` moves the "containing area" to the children only so we can't assert anything on the target collection ("NEW_COLLECTION") in this case. using `.parent()` once again moves us to the top level container for this scope - we can see both the target collection and its children at once.
```javascript
function getSidebarCollectionChildren(item) {
  return cy
    .findByTestId("sidebar")
    .findByText(item)
    .closest("a")
    .parent()
    .parent();
}
```

By doing this, we can finally make assertion like this one:
```javascript
getSidebarCollectionChildren(NEW_COLLECTION).within(() => {
  cy.icon("chevrondown").should("have.length", 2); // both target collection and "First collection" are open
  cy.findByText("First collection");
  cy.findByText("Second collection");
});
```

And yes, I agree we should have some comment explaining what's going on here. Whenever we need to use some obscure and complicated selector that requires a comment to even understand it, that is a good pointer that HTML structure can be improved. Following the guidelines of React Testing Library that encourages good accessibility practices, we can _at least_ use `aria-expanded` attribute and that would've already made our life easier for testing.
Not saying you should implement this in this commit. Just analyzing the problem and thinking out loud.

Another thing we could've solved with pure CSS (had we used BEM, for example) is some class that clearly denotes top level group/container for collections that contain children.

Reply:
Awesome, thanks for your suggestions, the selector looks much better now. I also added `tree` and `treeitem` roles to the sidebar and its children, and an `aria-expanded` prop to sidebar children. So we can now use the `findByRole` selector when testing the sidebar